javascript/angular-todo-davinci/src/app/app.component.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * angular-todo-prototype
+ *
+ * app.component.ts
+ *
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+import { OnInit, inject } from '@angular/core';
+import { Component } from '@angular/core';
+import { Config, UserManager } from '@forgerock/javascript-sdk';
+import { AsyncConfigOptions } from '@forgerock/javascript-sdk/src/config/interfaces';
+import { UserService } from './services/user.service';
+import { Router, RouterOutlet } from '@angular/router';
+import { NavigationCancel, NavigationEnd, NavigationError, NavigationStart } from '@angular/router';
+import { Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import createConfig from '../utilities/create-config';
+
+import { LoadingComponent } from './utilities/loading/loading.component';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  standalone: true,
+  imports: [RouterOutlet, LoadingComponent],
+})
+export class AppComponent implements OnInit {
+  private readonly userService = inject(UserService);
+  private readonly router = inject(Router);
+
+  loading = false;
+
+  constructor() {
+    const router = this.router;
+
+    const navStart = router.events.pipe(
+      filter((evt) => evt instanceof NavigationStart),
+    ) as Observable<NavigationStart>;
+
+    const navEnd = router.events.pipe(
+      filter(
+        (evt) =>
+          evt instanceof NavigationEnd ||
+          evt instanceof NavigationCancel ||
+          evt instanceof NavigationError,
+      ),
+    );
+
+    navStart.subscribe(() => (this.loading = true));
+    navEnd.subscribe(() => (this.loading = false));
+  }
```

Comment:
@cerebrl @ryanbas21 what do you think of removing this constructor and the loading component in the app's html template? The loading message is `Verifying access` which this constructor doesn't really do. We do attempt to check for authorization in the ngOnInit but even there I think we can load silently.